text/0030-protocol-messages.md

```diff
@@ -0,0 +1,248 @@
++ Feature name: `protocol-messages`
++ Start date: 2020-04-27
++ RFC PR: [iotaledger/bee-rfcs#30](https://github.com/iotaledger/bee-rfcs/pull/30)
++ Bee issue: [iotaledger/bee#70](https://github.com/iotaledger/bee/issues/70)
+
+# Summary
+
+This RFC introduces the IOTA protocol messages that were initially added in
+[IRI#1393](https://github.com/iotaledger/iri/pull/1393).
+
+# Motivation
+
+To be able to take part in the IOTA networks, Bee nodes need to implement the exact same protocol presented in this RFC
+and currently being used by [IRI](https://github.com/iotaledger/iri) nodes and
+[HORNET](https://github.com/gohornet/hornet) nodes. However, it does not necessarily mean implementing the same versions
+of the protocol and a design decision - later explained - concludes that Bee nodes and IRI nodes will not be able to
+communicate with each other.
+
+# Detailed design
+
+This section details:
+- The `Message` trait that provides serialization and deserialization of messages to and from byte buffers;
+- A type-length-value protocol - on top of the trait - that adds metadata in order to send and receive the messages over
+  a transport layer;
+- The current `Message` implementations representing handshake, requests, responses, events, ...;
+
+## `Message` trait
+
+The `Message` trait is protocol agnostic and only provides serialization and deserialization to and from byte buffers.
+It should not be used as is but rather be paired with a higher layer - like a type-length-value encoding - and as such
+does not provide any safety check on inputs/outputs.
```

Comment:
In this context, what does 'safety' mean? Does it mean that Rust's safety guarantees may be invalidated with bad inputs? If so, the `from_bytes` method should probably be declared `unsafe`.

Reply:
I meant that the bounds of the slice are not checked

Reply:
If an invalid size is given it may panic